pulpcore/cli/common/openapi.py

```diff
@@ -164,54 +170,117 @@ def extract_params(
-                        assert isinstance(param, int)
+                    param = self.validate_schema(param_schema, name, param)
+
+                if isinstance(param, List):
+                    # Check if we need to implode the list
+                    style = (
+                        param_spec.get("style") or "form"
+                        if param_in in ("query", "cookie")
+                        else "simple"
+                    )
+                    explode: bool = param_spec.get("explode", style == "form")
+                    if not explode:
+                        # Not exploding means comma separated list
+                        param = ",".join(param)
                 result[name] = param
         remaining_required = [
             item["name"] for item in param_specs.values() if item.get("required", False)
         ]
         if any(remaining_required):
             raise RuntimeError(
-                _("Required parameters [{required}] missing for {param_type}.").format(
-                    required=", ".join(remaining_required), param_type=param_type
+                _("Required parameters [{required}] missing in {param_in}.").format(
+                    required=", ".join(remaining_required), param_in=param_in
                 )
             )
         return result
 
-    def validate_body(self, schema: Any, body: Dict[str, Any], uploads: Dict[str, bytes]) -> None:
+    def validate_schema(self, schema: Any, name: str, value: Any) -> Any:
+        # Look if the schema is provided by reference
         schema_ref = schema.get("$ref")
         if schema_ref:
             if not schema_ref.startswith("#/components/schemas/"):
                 raise OpenAPIError(_("Api spec is invalid."))
             # len("#/components/schemas/") == 21
             schema_name = schema_ref[21:]
             schema = self.api_spec["components"]["schemas"][schema_name]
-        for key, value in body.items():
-            field_spec = schema["properties"].get(key)
-            if not field_spec:
-                raise OpenAPIError(_("Unexpected field '{key}' provided.").format(key=key))
+
+        if value is None and schema.get("nullable", False):
+            return None
+
+        schema_type = schema.get("type", "string")
+        if schema_type == "object":
+            value = self.validate_object(schema, name, value)
+        elif schema_type == "array":
+            value = self.validate_array(schema, name, value)
+        elif schema_type == "string":
+            value = self.validate_string(schema, name, value)
+        elif schema_type == "integer":
+            value = self.validate_integer(schema, name, value)
+        elif schema_type == "boolean":
+            value = self.validate_integer(schema, name, value)
+            if not isinstance(value, bool):
+                raise OpenAPIError(_("'{name}' is expected to be a boolean.").format(name=name))
+        # TODO: Add more types here.
+        return value
```

Comment:
Can we add a check for floats/doubles? https://swagger.io/specification/#data-types